web/src/settings_data.ts

```diff
@@ -298,6 +298,10 @@ export function bot_type_id_to_string(type_id: number): string | undefined {
 }
 
 export function user_can_access_all_other_users(): boolean {
+    if (page_params.is_spectator) {
+        return true;
+    }
```

Comment:
This deserves a comment explaining why this one needs to be hardcoded differently. Something like:

> // While spectators have is_guest=true for convenience in some code paths, they do not currently use the guest user systems for limiting their user access to subscribers of web-public channels, which is typically the entire user set for a server anyway.

Reply:
Fixed!